algebra-core/src/curves/models/twisted_edwards_extended.rs

```diff
@@ -165,6 +165,129 @@ impl<P: Parameters> AffineCurve for GroupAffine<P> {
     }
 }
 
+impl<P: Parameters> BatchGroupArithmetic for GroupAffine<P> {
+    type BBaseField = P::BaseField;
+
+    fn batch_double_in_place(
+        bases: &mut [Self],
+        index: &[usize],
+        _scratch_space: Option<&mut Vec<Self::BBaseField>>,
+    ) {
+        Self::batch_add_in_place(
+            bases,
+            &mut bases.to_vec()[..],
+            &index
+                .iter()
+                .map(|&x| (x, x))
+                .collect::<Vec<(usize, usize)>>()[..],
+        );
+    }
+
+    // Total cost: 12 mul. Projective formulas: 11 mul.
+    fn batch_add_in_place_same_slice(bases: &mut [Self], index: &[(usize, usize)]) {
```

Comment:
Seems like a lot of shared code between this method and the next. Can we dedup somehow?

Reply:
I noticed that one can indeed dedup. I used macros as defining a private trait was to cumbersome

Reply:
I think macros is a good way, do you want to do that change?

Reply:
I've just seen you did, thanks!